subprojects/gradle/tms/src/main/kotlin/com/avito/plugin/MarkReportAsSourceAction.kt

```diff
@@ -19,42 +19,46 @@ class MarkReportAsSourceAction(
     private val timeProvider: TimeProvider,
     private val logger: Logger
 ) {
+    private val tag = "TMS"
 
     fun mark(reportCoordinates: ReportCoordinates) {
         val testSuiteVersion = timeProvider.nowInMillis()
 
-        logger.debug("This is a new version [$testSuiteVersion] of full test suite for tms")
+        when (val result = reportsApi.getReport(reportCoordinates)) {
+            is GetReportResult.Found -> {
 
-        val reportId = tryGetId(reportCoordinates)
-
-        if (reportId != null) {
-            reportsApi.pushPreparedData(
-                reportId = reportId,
-                analyzerKey = "test_suite",
-                preparedData = jsonObject(
-                    "full" to true,
-                    "version" to testSuiteVersion
-                )
-            ).onFailure { error ->
-                logger.critical("Can't push prepared data: testSuite info", error)
-            }
-
-            reportsApi.setFinished(reportCoordinates).onFailure { error ->
-                logger.critical("Can't finish report for coordinates: $reportCoordinates", error)
+                reportsApi.pushPreparedData(
+                    reportId = result.report.id,
+                    analyzerKey = "test_suite",
+                    preparedData = jsonObject(
+                        "full" to true,
+                        "version" to testSuiteVersion
+                    )
+                ).fold(
+                    {
+                        reportsApi.setFinished(reportCoordinates).fold(
+                            {
+                                logger.info(
+                                    "[$tag] Test suite for tms version $testSuiteVersion, " +
```

Comment:
do we need tags like that? it's not straightforward to me how to filter build log for TMS messages only

Reply:
CILogger already appends a prefix. Isn't that enough?

Reply:
stdout destination doesn't print any tag as i see; we should handle it; but not in this PR